sdks/python/apache_beam/ml/inference/sklearn_inference.py

```diff
@@ -183,6 +195,18 @@ def run_inference(
     splits = [
         vectorized_batch.iloc[[i]] for i in range(vectorized_batch.shape[0])
     ]
+    if isinstance(predictions, dict):
+      # Go from one dictionary of type: {key_type1: Iterable<value_type1>,
+      # key_type2: Iterable<value_type2>, ...} where each Iterable is of
+      # length batch_size, to a list of dictionaries:
+      # [{key_type1: value_type1, key_type2: value_type2}]
+      predictions_per_split = [
+          dict(zip(predictions, v)) for v in zip(*predictions.values())
+      ]
+      return [
+          PredictionResult(example, inference) for example,
+          inference in zip(splits, predictions_per_split)
+      ]
```

Comment:
Similarly to the Pytorch file, could we refactor this into a helper? Though it looks a bit more difficult since we're working with different types (numpy vs pandas) now, so I don't think we have to. It may make sense to explicitly keep the logic here to clarify. I'll leave this up to you.

Reply:
I think it can be refactored cleanly - done